Automattic/wp-calypso
# Save Site Profiler state to redux to persist state when changing routes

Fixes https://github.com/Automattic/dotcom-forge/issues/9244

## Proposed Changes

* Fixed retest button
* Save hash to redux
* Save page state to redux
* Adjusted the steps so that it makes sense

https://github.com/user-attachments/assets/bbdc5c46-d46c-41b2-b093-41a9028184e8

Note: The progress steps were all "fake," probably for UX reasons, as the API endpoint only has three statuses: `queued`, `running`, and `completed`.

## Why are these changes being made?


* Better user experience

## Testing Instructions



* Go to `calypso.localhost:3000/sites/performance/:site`
* Run a test, wait for the first step to complete
* Switch to another tab
* Switch back to the tab and the test should not let you feel like it's starting over
* Check nothing is broken at `calypso.localhost:3000/speed-test-tool?url=<site>`

Remember that this is a "perceived" progress, so acceptance criteria is as long as the user experience "feels smooth".

## Pre-merge Checklist
